ion/src/main/java/com/fasterxml/jackson/dataformat/ion/IonParser.java

```diff
@@ -397,7 +397,15 @@ public NumberType getNumberType() throws IOException
                 //Ion decimals can be arbitrary precision, need to read as big decimal
                 return NumberType.BIG_DECIMAL;
             case INT:
-                IntegerSize size = _reader.getIntegerSize();
+                IntegerSize size = null;
+                try {
+                    size = _reader.getIntegerSize();
+                } catch (NullPointerException e) {
+                    // Possible exception
+                }
```

Comment:
Note: this try/catch will no longer needed once https://github.com/amazon-ion/ion-java/issues/685 is addressed.

Reply:
Thanks for the note. I have added a comments mentioning it is a temporary fix on jackson until amazon-ion/ion-java#685 is addressed and published.